crates/lib/src/qvm/api.rs

```diff
@@ -0,0 +1,376 @@
+//! This module provides types and functions for making API calls directly to the QVM.
+//! Consider [`super::run_program`] for higher level access to the QVM that allows
+//! for running parameterized programs.
+use std::collections::HashMap;
+
+use qcs_api_client_common::ClientConfiguration;
+use reqwest::Response;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+
+use crate::RegisterData;
+
+use super::Error;
+
+#[derive(Serialize, Copy, Clone, Debug, Ord, PartialOrd, Eq, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+enum RequestType {
+    Multishot,
+    MultishotMeasure,
+    Expectation,
+    Wavefunction,
+}
+
+#[derive(Debug, Deserialize, Clone, Eq, PartialEq)]
+pub(super) struct Failure {
+    /// The message from QVM describing what went wrong.
+    pub(super) status: String,
+}
+
+/// A QVM response that can be deserialized into some successful response type `T`, or into a
+/// [`Failure`] containing an error message.
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+#[serde(untagged)]
+pub(super) enum QvmResponse<T>
+where
+    T: DeserializeOwned,
+{
+    #[serde(bound = "")]
```

Comment:
This is needed for generic types that are only bounded by `DeserializeOwned` to be recognized as having `Deserialize`. See [serde#1296](https://github.com/serde-rs/serde/issues/1296).